lib/addons/amp/index.ts

```diff
@@ -0,0 +1,150 @@
+    SIDECAR = 'sidecar'
+}
+
+/**
+ * Defaults options for the add-on
+ */
+const defaultProps = {
+    workspaceName: 'blueprints-amp-workspace',
+    workspaceTags: [{
+        key: 'Name',
+        value: 'blueprints-amp-workspace',
+      },
+      {
+        key: 'Environment',
+        value: 'blueprints-sandbox',
+      }
+    ],
+    deploymentMode: DeploymentMode.DEPLOYMENT,
+    namespace: 'default'
+};
+
+/**
+ * Implementation of AMP add-on for EKS Blueprints. Installs ADOT Collector.
+ */
+export class AmpAddOn implements ClusterAddOn {
+
+    readonly ampAddOnProps: AmpAddOnProps;
+
+    constructor(props?: AmpAddOnProps) {
+        this.ampAddOnProps = { ...defaultProps, ...props };
+    }
+
+    @dependable(AdotCollectorAddOn.name)
+    deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+        const cluster = clusterInfo.cluster;
+        let finalRemoteWriteURLEndpoint: string;
+        let finalDeploymentMode: string;
+        let finalNamespace: string;
+        let doc: string;
+        let cfnWorkspace: aps.CfnWorkspace|undefined;
+        
+        let cfnWorkspaceProps : CfnWorkspaceProps  = {
+            alias: defaultProps.workspaceName,  
+            tags: defaultProps.workspaceTags
+        };
+        if (this.ampAddOnProps.prometheusRemoteWriteURL) {
+            finalRemoteWriteURLEndpoint = this.ampAddOnProps.prometheusRemoteWriteURL;
+        }
+        else {
+            if (this.ampAddOnProps.workspaceName){
+                cfnWorkspaceProps = {
+                    ...cfnWorkspaceProps,
+                    alias:this.ampAddOnProps.workspaceName
+                };        
+            }
+    
+            if (this.ampAddOnProps.workspaceTags){
+                cfnWorkspaceProps = {
+                    ...cfnWorkspaceProps,
+                    tags:this.ampAddOnProps.workspaceTags
+                }; 
+            }
+            cfnWorkspace = new aps.CfnWorkspace(cluster.stack, 'MyAMPWorkspace', cfnWorkspaceProps);/* all optional props */ 
+            const URLEndpoint = cfnWorkspace.attrPrometheusEndpoint;
+            finalRemoteWriteURLEndpoint = URLEndpoint + 'api/v1/remote_write';
+        }
+
+        finalDeploymentMode = defaultProps.deploymentMode;
+        if (this.ampAddOnProps.deploymentMode) {
+            finalDeploymentMode = this.ampAddOnProps.deploymentMode;
+        }
+
+        finalNamespace = defaultProps.namespace;
+        if (this.ampAddOnProps.namepace) {
+            finalNamespace = this.ampAddOnProps.namepace;
+        }
+
+        // Applying manifest for configuring ADOT Collector for Amp.
+        if (finalDeploymentMode == DeploymentMode.DAEMONSET) {
+            doc = readYamlDocument(__dirname + '/collector-config-amp-daemonset.ytpl');
+        }
+        else {
+            doc = readYamlDocument(__dirname + '/collector-config-amp.ytpl');
+        }
+        const docArray = doc.replace(/{{your_remote_write_endpoint}}/g, finalRemoteWriteURLEndpoint)
```

Comment:
We have [`KubectlProvider.addManifest`](https://github.com/aws-quickstart/cdk-eks-blueprints/blob/main/lib/addons/helm-addon/kubectl-provider.ts#L144)  - it seems to be doing the same thing that you are doing. Please take a look and let's discuss.